File: easy/1624_largest_substring_between_two_equal_characters/solution.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        c_map = {}

        max_length = -1

        for i, c in enumerate(s):
            if c in c_map:
                max_length = max(max_length, i - c_map[c] - 1)
            else:
                c_map[c] = i

        return max_length

File: easy/1624_largest_substring_between_two_equal_characters/solution.cpp
#include <vector>

class Solution {
public:
  int maxLengthBetweenEqualCharacters(std::string s) {
    std::vector<int> cArray(26, -1);

    int maxLength = -1;

    for (int i = 0; i < s.size(); ++i) {
      if (cArray[s[i] - 'a'] != -1)
        maxLength = std::max(maxLength, i - cArray[s[i] - 'a'] - 1);
      else
        cArray[s[i] - 'a'] = i;
    }

    return maxLength;
  }
};
